sos/plugins/juju.py

```diff
@@ -76,37 +86,40 @@ def export_mongodb(self):
                 suggest_filename="{}.json".format(collection))
 
     def setup(self):
-        self.add_copy_spec([
-            "/var/lib/juju"
-        ])
         limit = self.get_option("log_size")
         self.add_copy_spec_limit("/var/log/upstart/juju-db.log",
                                  sizelimit=limit)
         self.add_copy_spec_limit("/var/log/upstart/juju-db.log.1",
                                  sizelimit=limit)
         if not self.get_option("all_logs"):
             # Capture the last bit of all files
-            for filename in os.listdir("/var/log/juju/"):
-                if filename.endswith(".log"):
-                    fullname = "/var/log/juju/" + filename
-                    self.add_copy_spec_limit(fullname, sizelimit=limit)
-            # Do just the all-machines from juju local
-            self.add_copy_spec_limit("/var/log/juju-*/all-machines.log",
-                                     sizelimit=limit)
+            if(os.path.isdir("/var/log/juju/")):
+                for filename in os.listdir("/var/log/juju/"):
+                    if filename.endswith(".log"):
+                        fullname = "/var/log/juju/" + filename
+                        self.add_copy_spec_limit(fullname, sizelimit=limit)
             self.add_cmd_output('ls -alRh /var/log/juju*')
+            self.add_cmd_output('ls -alRh /var/lib/juju/*')
+
         else:
             self.add_copy_spec([
                 "/var/log/juju",
-                "/var/log/juju-*"
+                "/var/log/juju-*",
+                "/var/lib/juju"
```

Comment:
What's in this directory? It could use a comment as locations in `/var` sometimes wind up with considerable amounts of data. If that's never going to be the case here as comment would be good to explain that.